### PR TITLE
Release 0.103.0

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Mainnet states match:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ go run scripts/compare-states/compare-states.go http://seed1.neo.org:10332 http://localhost:10332
at 0: d34893828a2ddb99c87226a92f764ae2dfadd0d2dc53fb33ae425be83b8d30b4 vs d34893828a2ddb99c87226a92f764ae2dfadd0d2dc53fb33ae425be83b8d30b4
at 4280945: 4686be77f36c6ad34e86c32ce03145cee4996fb445696d4f2155aeb6234b9f39 vs 4686be77f36c6ad34e86c32ce03145cee4996fb445696d4f2155aeb6234b9f39
```
Testnet states match:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ go run scripts/compare-states/compare-states.go http://seed1t5.neo.org:20332 http://localhost:20332
at 0: ed359c92fd2e5d3fb6cc57c1599481b223b9eea82bcf940dd1a3db354a9b99d6 vs ed359c92fd2e5d3fb6cc57c1599481b223b9eea82bcf940dd1a3db354a9b99d6
at 2877018: 8ea0bec3b0e679c1cb09542911f21dd637f878bfe01ce8b3f5fc1c31a37d13e8 vs 8ea0bec3b0e679c1cb09542911f21dd637f878bfe01ce8b3f5fc1c31a37d13e8
```

### CODE REVIEW

**Comment by AnnaShaleva** [comment]
Let's update interop dependenciew before the release, just in case. The latest gnark update made by dependabot didn't include it.

**Comment by AnnaShaleva** [comment]
Also added a note about #3110 in the "Bugs fixed" section via the last force-push.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,75 @@
 
 This document outlines major changes between releases.
 
+## 0.103.0 "Buckminsterfullerene" (20 Oct 2023)
+
+A major 3.6.0-compatible version of NeoGo with new salty ZKP-related
```
Comment (roman-khimov) [inline CHANGELOG.md:7]: minor, in X.Y.Z X is major while Y is minor.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,75 @@
...
+`TimePerBlock`, a set of node and services address- and port- related configuration
+section that were replaced by the new `Addresses` format, direct `UnlockWallet`
+consensus configuration that was replaced by a separate `Consensus` section and
+a set of node-specific protocol configurations that were moved under a separate
+`P2P` section.
+
+This release fixes a set of bugs including a vulnerability introduced by changes
+in the transaction conflicts storage scheme implemented in #3061 (a part of
+0.102.0 release). With the patch presented, it's impossible to uncontrollably
+pollute the storage with malicious conflicting records.
```
Comment (roman-khimov) [inline CHANGELOG.md:29]: Probably a bit more text than I'd expect. But anyway.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,75 @@
...
+* database scheme is changed by new way of storing the transaction conflicting
+  records (#3138) 
+
+Improvements:
+* NeoFS SDK dependency upgrade (#3129)
+* gnark and gnark-crypto dependencies upgrade (#3145, #3162, #3163)
+* introduce timeout restriction for BoltDB opening (#3148)
+* bump code coverage uploading action version (#3153)
+* Go 1.21 support, bump minimum required Go version up to Go 1.19 (#3157)
+* upgrade golang.org/x/net version
```
Comment (roman-khimov) [inline CHANGELOG.md:55]: No PR link

File: CHANGELOG.md (hunk 2 above)
Comment (AnnaShaleva) [reply CHANGELOG.md:29]: Is it important, maybe remove this paragraph at all?

File: CHANGELOG.md (hunk 3 above)
Comment (AnnaShaleva) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,72 @@
 
 This document outlines major changes between releases.
 
+## 0.103.0 "Buckminsterfullerene" (20 Oct 2023)
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply CHANGELOG.md]: Fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (roman-khimov) [reply CHANGELOG.md:29]: I mean whole text, but we can keep it as is.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**